autopilot/contractor.go

```diff
@@ -313,8 +316,14 @@ func (c *contractor) runContractChecks(ctx context.Context, cfg api.AutopilotCon
 			continue
 		}
 
+		// if the host is blocked we ignore it, it might be unblocked later
+		if host.Blocked {
```

Comment:
Is there a reason for now moving this into `isUsableHost`? Seems reasonable to exclude blocked hosts from other things such as determining the min score as well.

Reply:
Yeah we reuse `isUsableHost` when we loop over all hosts, which already exclude all blocked hosts. I only want to return `HostInfo` when we are fetching one host because I have to preload the blocklist and allowlist